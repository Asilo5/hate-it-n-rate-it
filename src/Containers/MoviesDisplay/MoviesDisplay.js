import React from 'react';
import './MoviesDisplay.scss';

const MoviesDisplay = ({ average_rating, id, backdrop_path, overview, poster_path, release_date, title}) => {
    return (
        <section className='movies-display'>
            <p>Avg. rating: {average_rating}</p>
            <p>{title}</p>
            <img src={poster_path} alt={title} />
        </section>
    )
}

export default MoviesDisplay;