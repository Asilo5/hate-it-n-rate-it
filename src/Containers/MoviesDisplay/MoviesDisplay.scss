.movies-display {
    width: 30%;
    margin: 28px;
    text-decoration: none;
    text-align: center;
    img {
        width: 80%;
    }
    &:hover {
        transform: scale(1.03);
    }
    h3 {
       
    }

} 

