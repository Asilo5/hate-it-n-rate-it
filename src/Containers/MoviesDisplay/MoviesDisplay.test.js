import React from 'react';
import { shallow } from 'enzyme';
import MoviesDisplay from './MoviesDisplay';

describe('MoviesDisplay', () => {
            let mockTitle = "Jumanji: The Next Level";
            let posterPath = "https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg"
            let backdropPath = "https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg"
            let releaseDate = "2019-12-04"
            let mockOverview = "In Jumanji: The Next Level, the gang is back but the game has changed."


    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<MoviesDisplay id={1} 
                        title={mockTitle}
                        poster_path={posterPath}
                        backdrop_path={backdropPath}    
                        release_date={releaseDate}
                        overview={mockOverview}
                        average_rating={4}
                        />)
    })

    it('should match snapshot with all information passing in correctly', () => {
      expect(wrapper).toMatchSnapshot();
    })
    
})