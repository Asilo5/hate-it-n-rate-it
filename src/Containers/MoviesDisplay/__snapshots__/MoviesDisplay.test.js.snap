// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesDisplay should match snapshot with all information passing in correctly 1`] = `
<Link
  className="movies-display"
  key="1"
  to="/movies/1"
>
  <section>
    <p>
      Avg. rating: 
      4
    </p>
    <h3>
      Jumanji: The Next Level
    </h3>
    <img
      alt="Jumanji: The Next Level"
      src="https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg"
    />
  </section>
</Link>
`;
